### PR TITLE
Adds some documentation changes and changes to PII interface

### PR DESCRIPTION
**Pull request by abhijitpal1247**
- This PR adds support for PII, for both input and output end
- Adds some more documentation for AutoGuard usage

### CODE REVIEW

File: nemoguardrails/library/autoguard/README.md [hunk 1]
```diff
@@ -4,92 +4,216 @@ This package implements the AutoGuard API integration.
...
+3. [Jailbreak Detection](#jailbreak-detection)
+4. [Confidential Detection](#confidential-detection)
 5. [Racial bias Detection](#racial-bias-detection)
-6. [Jailbreak Detection](#jailbreak-detection)
-7. Factcheck
-8. PII
+6. [Tonal Detection](#tonal-detection)
+7. [Toxicity detection](#toxicity-extraction)
+8. [PII](#pii)
+9. [Factcheck](#factcheck)
```
Comment (rahm-decypher) [inline nemoguardrails/library/autoguard/README.md:16]: @abhijitpal1247 we are till missing the IP detection guardrail? please add that in

File: nemoguardrails/library/autoguard/README.md [hunk 2]
```diff
@@ -4,92 +4,216 @@ This package implements the AutoGuard API integration.
...
+
+
+Note: Factcheck and PII are implemented a bit differently, compared to other guardrails.
```
Comment (rahm-decypher) [inline nemoguardrails/library/autoguard/README.md:19]: can you go through the api doc here https://docs.google.com/document/d/1B4_5JZ0BcX9y95DMULMftripYTWh5z-Nh89pfWgYJAE/edit?usp=sharing and see if you can add any guardrail configuration info/language (especially for pii/fact checking) that may not be present in this readme?

File: nemoguardrails/library/autoguard/README.md [hunk 3]
```diff
@@ -98,82 +222,91 @@ rails:
...
-            "[TRANSACTION_ID]": 0.5,
-            "[RELIGION]": 0.5,
-        }}
-  input:
-    flows:
-      - call autoguard pii
+        endpoint: "https://nvidia.autoalign.ai/guardrail"
+      output:
+        entities:
+          - '[BANK ACCOUNT NUMBER]'
```
Comment (rahm-decypher) [inline nemoguardrails/library/autoguard/README.md:242]: are these the latest types after @Santhanreddy71 de-duped a bunch?

File: nemoguardrails/library/autoguard/actions.py [hunk 4]
```diff
@@ -12,11 +12,10 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import asyncio
```
Comment (rahm-decypher) [inline nemoguardrails/library/autoguard/actions.py]: why is this needed?

File: tests/test_autoguard.py [hunk 5]
```diff
@@ -152,22 +136,14 @@ async def test_harm_to_human_input():
...
-    async def mock_autoguard_output_api(context: Optional[dict] = None, **kwargs):
-        query = context.get("bot_message")
-        if (
-            query
-            == "Harm to human violation has been detected by AutoGuard; Sorry, can't process."
-        ):
-            return False, None
+            return (
+                True,
+                "Harm to human violation has been detected by AutoGuard; Sorry, can't process.",
```
Comment (rahm-decypher) [inline tests/test_autoguard.py]: change it to "Potential harm to human has been detected by AutoGuard; Sorry, can't process."

**Review by rahm-decypher — CHANGES_REQUESTED** [review]
@KimiJL please review it one more time tomorrow before we merge. thanks

**Review by KimiJL — APPROVED** [review]
Looks fine on my end, should be good if QA is good

File: nemoguardrails/library/autoguard/README.md (hunk 1 above)
Comment (abhijitpal1247) [reply nemoguardrails/library/autoguard/README.md:16]: resolved this issue

File: nemoguardrails/library/autoguard/actions.py (hunk 4 above)
Comment (abhijitpal1247) [reply nemoguardrails/library/autoguard/actions.py]: resolved this issue

File: tests/test_autoguard.py (hunk 5 above)
Comment (abhijitpal1247) [reply tests/test_autoguard.py]: resolved this issue

**Review by rahm-decypher — APPROVED** [review]
